### PR TITLE
[Model] Multi-input support for LLaVA and fix embedding inputs for multi-image models

### LINKED ISSUE #8236: [Feature]: Add multi-image input support for LLaVA offline inference (similar to #7230)
### 🚀 The feature, motivation and pitch

Recently, I released a model https://huggingface.co/aws-prototyping/long-llava-qwen2-7b using the `llava` model architecture and could support multiple images (or video) as input, so it is reasonable to enable multiple images as a input for llava in vllm also. Thank you!

### Alternatives

_No response_

### Additional context

_No response_

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

### PR DESCRIPTION
**Pull request by DarkLight1337**
This PR extends the vLLM support for multi-image input from LLaVA-NeXT to LLaVA. Note that LLaVA from the original authors hasn't been trained on multi-image inputs, so this is mostly useful for custom trained models.

I also found and fixed a hidden bug where embeddings cannot be inputted for models that support multi-image inputs.

FIX #8236

### CODE REVIEW

File: docs/source/models/supported_models.rst [hunk 1]
```diff
@@ -219,14 +219,19 @@ Multimodal Language Models
...
   * - :code:`LlavaNextForConditionalGeneration`
     - LLaVA-NeXT
     - Image\ :sup:`E+`
     - :code:`llava-hf/llava-v1.6-mistral-7b-hf`, :code:`llava-hf/llava-v1.6-vicuna-7b-hf`, etc.
     -
+  * - :code:`MiniCPMV`
+    - MiniCPM-V
+    - Image\ :sup:`+`
+    - :code:`openbmb/MiniCPM-V-2` (see note), :code:`openbmb/MiniCPM-Llama3-V-2_5`, :code:`openbmb/MiniCPM-V-2_6`, etc.
+    -
```
Comment (DarkLight1337) [inline docs/source/models/supported_models.rst:234]: This fixes an issue where the list isn't in alphabetical order.

File: docs/source/models/supported_models.rst [hunk 2]
```diff
@@ -237,14 +242,9 @@ Multimodal Language Models
...
-  * - :code:`MiniCPMV`
-    - MiniCPM-V
-    - Image\ :sup:`+`
-    - :code:`openbmb/MiniCPM-V-2` (see note), :code:`openbmb/MiniCPM-Llama3-V-2_5`, :code:`openbmb/MiniCPM-V-2_6`, etc.
-    -
   * - :code:`QWenLMHeadModel`
-    - Qwen
-    - Image
+    - Qwen-VL
+    - Image\ :sup:`E`
```
Comment (DarkLight1337) [inline docs/source/models/supported_models.rst:247]: This updates the Qwen-VL row to follow the new format.

File: tests/conftest.py [hunk 3]
```diff
@@ -671,7 +671,7 @@ def generate_greedy(
         self,
         prompts: List[str],
         max_tokens: int,
-        images: Optional[List[Image.Image]] = None,
+        images: Optional[PromptImageInput] = None,
```
Comment (DarkLight1337) [inline tests/conftest.py:674]: The changes in this file are to resolve type errors when passing multi-input.

File: tests/models/test_llava.py [hunk 4]
```diff
@@ -181,6 +253,51 @@ def test_models(hf_runner, vllm_runner, image_assets, model, size_factors,
...
+        vllm_runner,
+        inputs,
+        model,
+        dtype=dtype,
+        max_tokens=max_tokens,
+        num_logprobs=num_logprobs,
+        tensor_parallel_size=1,
+    )
+
+
```
Comment (DarkLight1337) [inline tests/models/test_llava.py:300]: Copying the LLaVA-NeXT test layout for now. The logic will be consolidated in a future PR.

**Review by ywang96 — APPROVED** [review]
LGTM!

**Pull request merged**